pkg/volume/secret/secret.go

```diff
@@ -188,12 +192,16 @@ func (b *secretVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
-	kubeClient := b.plugin.host.GetKubeClient()
-	if kubeClient == nil {
-		return fmt.Errorf("Cannot setup secret volume %v because kube client is not configured", b.volName)
+	var secret *v1.Secret
+	if b.getSecret != nil {
+		secret, err = b.getSecret(b.pod.Namespace, b.source.SecretName)
+	} else {
+		kubeClient := b.plugin.host.GetKubeClient()
```

Comment:
Agreed. Why can't `getSecret` function replace this? I mentioned the same thing in the previous PR (#39558) too...